src/components/cardboard-controls.js

```diff
@@ -0,0 +1,34 @@
+const CARDBOARD_BUTTON = "Cardboard Button";
+
+module.exports = AFRAME.registerComponent("cardboard-controls", {
+  init: function() {
+    this.buttons = {};
+    this.gamepad = null;
+  },
+
+  tick: function() {
+    this.gamepad = this.gamepad || this._getGamepad();
+    if (this.gamepad) {
+      for (let i = 0; i < this.gamepad.buttons.length; i++) {
+        if (this.gamepad.buttons[i].pressed && !this.buttons[i]) {
+          this.el.emit("cardboardbuttondown", {});
+        } else if (!this.gamepad.buttons[i].pressed && this.buttons[i]) {
+          this.el.emit("cardboardbuttonup", {});
+        }
+        this.buttons[i] = this.gamepad.buttons[i].pressed;
+      }
+    } else if (Object.keys(this.buttons)) {
+      this.buttons = {};
+    }
+  },
+
+  _getGamepad: function() {
+    const gamepads = navigator.getGamepads && navigator.getGamepads();
```

Comment:
I think you should just bail and return `null` if `navigator.getGamepads` is not truthy